### PR TITLE
Feature bl 5900 motr  allow multiple valid uk mobile number formats

### PR DESCRIPTION
**Pull request by matthewdoreilly**

### CODE REVIEW

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java [hunk 1]
```diff
@@ -76,11 +83,18 @@ public Response phoneNumberPageGet() throws Exception {
     @POST
     public Response phoneNumberPagePost(@FormParam("phoneNumber") String phoneNumber) throws Exception {
 
+        phoneNumber = trimWhitespace(phoneNumber);
+
         if (validator.isValid(phoneNumber)) {
```
Comment (djeley) [inline webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: Would it be easier to validate the normalised number?

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java [hunk 2]
```diff
@@ -99,4 +113,20 @@ public Response phoneNumberPagePost(@FormParam("phoneNumber") String phoneNumber
...
+        return phoneNumber.replaceAll("\\s+","");
+    }
+
+    private String normaliseUkPhoneNumber(String phoneNumber) {
+        PhoneNumberUtil phoneNumberUtil = PhoneNumberUtil.getInstance();
+        Phonenumber.PhoneNumber ukNumberProto = null;
+        try {
+            ukNumberProto = phoneNumberUtil.parse(phoneNumber, GB_REGION_CODE);
+        } catch (NumberParseException e) {
+            System.err.println("NumberParseException was thrown: " + e.toString());
```
Comment (djeley) [inline webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: Could we use the logger here? If this error is caught and logged, what will happen next?

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java [hunk 3]
```diff
@@ -99,4 +113,20 @@ public Response phoneNumberPagePost(@FormParam("phoneNumber") String phoneNumber
 
         return Response.ok(renderer.render(PHONE_NUMBER_TEMPLATE, modelMap)).build();
     }
+
+    private String trimWhitespace(String phoneNumber) {
+        return phoneNumber.replaceAll("\\s+","");
```
Comment (raffcecco) [inline webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: Minor point, but the code style for MOTR is a blank line between method signature and the body.

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java [hunk 4]
```diff
@@ -99,4 +113,20 @@ public Response phoneNumberPagePost(@FormParam("phoneNumber") String phoneNumber
...
+    }
+
+    private String normaliseUkPhoneNumber(String phoneNumber) {
```
Comment (djeley) [inline webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: Would prefer to see this method outside of the resource, e.g. in a utility class.

File: webapp/src/main/java/uk/gov/dvsa/motr/web/validator/PhoneNumberValidator.java [hunk 5]
```diff
@@ -31,21 +31,47 @@ public PhoneNumberValidator(SubscriptionsValidationService subscriptionsValidati
...
 
             return false;
         }
 
+        return true;
+    }
+
+    private boolean isPhoneNumberAValidUkNumber(String phoneNumber) {
+
+        Pattern validationRegex = Pattern.compile("^(\\(?07\\d{3}\\)?|\\+447\\d{3}|447\\d{3}|00447\\d{3}|\\(44\\)7\\d{3}|" +
```
Comment (djeley) [inline webapp/src/main/java/uk/gov/dvsa/motr/web/validator/PhoneNumberValidator.java]: If we normalised before validating, couldn't this regex be simplified to what it was previously?

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java (hunk 2 above)
Comment (matthewdoreilly) [reply webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: Done :thumbsup:

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java (hunk 4 above)
Comment (matthewdoreilly) [reply webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: I was considering that but didn't as it's only used in one place. May be useful in future, and good standard to keep this separate from the resource, so I've done that 👍

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java (hunk 3 above)
Comment (matthewdoreilly) [reply webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: 👍

File: webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java (hunk 1 above)
Comment (matthewdoreilly) [reply webapp/src/main/java/uk/gov/dvsa/motr/web/resource/PhoneNumberResource.java]: Doesn't make it easier or harder, but makes the code a bit clearer. Done 👍

**Pull request merged**